tools/rum-models-generator/Sources/RUMModelsGeneratorCore/RUM/RUMSwiftTypeTransformer.swift

```diff
@@ -136,6 +147,55 @@ internal class RUMSwiftTypeTransformer: TypeTransformer<SwiftType> {
             fixedName = format(structName: parentTypeName) + fixedName
         }
 
+        if fixedName == "Connectivity" {
+            fixedName = "RUMConnectivity"
+        }
+
+        if fixedName == "USR" {
+            fixedName = "RUMUser"
+        }
```

Comment:
i don't understand why we needed these, don't we already add `RUM` prefix to top level types? if `Connectivity` and `USR` are nested types: we don't need `RUM` prefix for them, do we?

Reply:
This is because of the way `RUM*Transformer` works - it first decorates root level types (so adds `RUM` prefix), then adds additional root level types (collected in `sharedRootTypes` array):
```swift
override func transform(types: [SwiftType]) throws -> [SwiftType] {
   sharedRootTypes = []
   let transformed = try types.map { try transformAny(type: $0) }
   return transformed + sharedRootTypes
}
```

Reply:
if that was a conscious decision then it's fine 👌 
personally i would expect to manage all top level types at once, like `if topLevel then addPrefix`

Reply:
Yes, this is intentional. Although this could be improved as you propose, I'll keep it like this until we have more than just 2 shared types to decorate.